hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ScmClient.java

```diff
@@ -52,4 +107,40 @@ public void setUpdateServiceGrpcClient(
   public SCMUpdateServiceGrpcClient getUpdateServiceGrpcClient() {
     return updateServiceGrpcClient;
   }
+
+  public Pipeline getContainerLocation(long containerId, boolean forceRefresh)
+      throws IOException {
+    if (forceRefresh) {
+      containerLocationCache.invalidate(containerId);
+    }
+    try {
+      return containerLocationCache.get(containerId);
+    } catch (ExecutionException e) {
+      return handleCacheExecutionException(e);
+    }
+  }
+
+  public Map<Long, Pipeline> getContainerLocations(Iterable<Long> containerIds,
```

Comment:
Is this for objects with multiple containers? Why have `getContainerLocation`?

Reply:
A key is designed to have multiple blocks and thus multiple containers. Indeed, we always use this interface and don't really need `getContainerLocation`. Removed.